packages/esm-patient-immunizations-app/src/immunizations/immunizations-form.component.tsx

```diff
@@ -183,36 +226,37 @@ const ImmunizationsForm: React.FC<DefaultWorkspaceProps> = ({ patientUuid, close
               type="text"
             />
           </DatePicker>
-        </div>
-        <div className={styles.immunizationSequenceSelect}>
+        </section>
+        <section className={styles.immunizationSequenceSelect}>
           <TextInput
             type="text"
             id="lotNumber"
             labelText={t('lotNumber', 'Lot Number')}
             value={formState.lotNumber}
             onChange={(evt) => updateSingle('lotNumber', evt.target.value)}
           />
-        </div>
-        <div className={styles.immunizationSequenceSelect}>
+        </section>
+        <section className={styles.immunizationSequenceSelect}>
           <TextInput
             type="text"
             id="manufacturer"
             labelText={t('manufacturer', 'Manufacturer')}
             value={formState.manufacturer}
             onChange={(evt) => updateSingle('manufacturer', evt.target.value)}
           />
-        </div>
-      </div>
+        </section>
+      </Stack>
       <ButtonSet className={isTablet ? styles.tablet : styles.desktop}>
         <Button className={styles.button} kind="secondary" onClick={() => closeWorkspace()}>
           {t('cancel', 'Cancel')}
         </Button>
         <Button
           className={styles.button}
           kind="primary"
-          disabled={isViewEditMode ? !enableEditButtons : !enableCreateButtons}
+          disabled={isViewEditMode ? !enableEditButton : !enableCreateButton}
+          type="submit"
         >
-          {t('save', 'Save')}
+          {t('saveAndClose', 'Save and close')}
```

Comment:
If you call this "Save and close" then the `handleFormSubmit` function should actually close the workspace, i.e., it needs to call `closeWorkspace()` at the appropriate point.